tsl/src/bgw_policy/drop_chunks_api.c

```diff
@@ -91,11 +220,67 @@ drop_chunks_add_policy(PG_FUNCTION_ARGS)
 				 errmsg("could not add drop_chunks policy because \"%s\" is not a hypertable",
 						get_rel_name(ht_oid))));
 
+	/* validate that the open dimension uses a time type */
+	open_dim = hyperspace_get_open_dimension(hypertable->space, 0);
```

Comment:
can this be refactored into a constructor for the interval type you created. The constructor should return the FormData_ts_interval or another struct representing the interval. It should live in the interval.c file you already created.